--- bluez-alsa-217cdde52b203798829242d8bea84871eec7fff2-src.orig/src/asound/bluealsa-pcm.c	2017-03-09 02:12:21.012875399 +0100
+++ bluez-alsa-217cdde52b203798829242d8bea84871eec7fff2-src/src/asound/bluealsa-pcm.c	2017-03-09 05:21:22.101299993 +0100
@@ -30,6 +30,8 @@
 /* Helper macro for obtaining the size of a static array. */
 #define ARRAY_SIZE(a) (sizeof(a) / sizeof((a)[0]))
 
+#define RATE_MAX (480000)
+#define CHANNELS_MAX 32U
 
 struct bluealsa_pcm {
 	snd_pcm_ioplug_t io;
@@ -469,15 +471,15 @@
 		return err;
 
 	if ((err = snd_pcm_ioplug_set_param_minmax(io, SND_PCM_IOPLUG_HW_PERIOD_BYTES,
-					128, 1024 * 4)) < 0)
+					128, 2 * 1024 * 1024)) < 0)
 		return err;
 
 	if ((err = snd_pcm_ioplug_set_param_minmax(io, SND_PCM_IOPLUG_HW_CHANNELS,
-					pcm->transport->channels, pcm->transport->channels)) < 0)
+					1, CHANNELS_MAX)) < 0)
 		return err;
 
 	if ((err = snd_pcm_ioplug_set_param_minmax(io, SND_PCM_IOPLUG_HW_RATE,
-					pcm->transport->sampling, pcm->transport->sampling)) < 0)
+					8000, RATE_MAX)) < 0)
 		return err;
 
 	return 0;
@@ -593,6 +595,7 @@
 		goto fail;
 
 	if ((ret = bluealsa_set_hw_constraint(pcm)) < 0) {
+	    debug("!!! hw_constraint !!!");
 		snd_pcm_ioplug_delete(&pcm->io);
 		goto fail;
 	}
@@ -601,6 +604,7 @@
 	return 0;
 
 fail:
+	debug("failed");
 	if (pcm->fd != -1)
 		close(pcm->fd);
 	if (pcm->event_fd != -1)
