--- flite-1.9.0-current.orig/src/audio/au_alsa.c	2017-03-09 03:53:31.502978016 +0100
+++ flite-1.9.0-current/src/audio/au_alsa.c	2017-03-09 04:18:12.976128004 +0100
@@ -271,9 +271,10 @@
   while (num_frames > 0) 
   {
 	res = snd_pcm_writei(pcm_handle, buf, num_frames);
-	if (res != num_frames) 
+	if (res <0) 
 	{
-	  if (res == -EAGAIN || (res > 0 && res < num_frames)) 
+	  cst_errmsg("audio_write_alsa: Error: %s.\n", snd_strerror(res));
+	  if (res == -EAGAIN) 
 	  {
 		snd_pcm_wait(pcm_handle, 100);
 	  }
